### ISSUE TITLE
Iterating over helper with each doesn't work with certain array data

### ISSUE THREAD

**Issue by satyavh** [ManuelDeLeon/viewmodel#136]
I'm migrating from Viewmodel 1 to 2 and run into a strange issue with helpers.
So I'm reading an array of base64 encoded images, store that in `@images()` that I'm displaying in the UI. With Viewmodel 1, I defined a helper `images` and used `each images`.

I haven't changed anything, but with Viewmodel 2 this doesn't work.

So I did some tests. If the array over which you iterate contains simple data, like `['a', 'b', 'c']` it works fine. But once you add base64 strings, they just get ignored.

Example, suppose `images` contains the following array:
```
0: Object
data: "hello"
1: Object
data: "hello"
2: Object
data: "data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wBDAAMCAgICAgMCAgIDAwMDBAYEBAQEBAgGBgUGCQgKCgkICQkKDA8MCgsOCwkJDRENDg8QEBEQCgwSExIQEw8QEBD/2wBDAQMDAwQDBAgEBAgQCwkLEBAQEBAQEBAQEANJI5B56mvVp4pNXTOCvlPZGVdWKxjEUqjHTiqL4QbWZeOOBWrM1iBlVkJHODz/ADqnOHlUrBaZAGScdK7KWKPJxGUNdDLmiO0u7oPxwayLlTuPkK5z3FdBFpjXBKyOB3Ix0pZbaGBTGhUDHfrXpYev7S1j5nHYL2V0cstrKCWkhJJ6e1SrbzzAL5Yx9MVsuLVGw0yjioTe2MWdsyA16VOV1ofMV009EQW+nRxYLnJ69Kt+WAR8v5VEl/aNy0w57+lWo57Nz8sqOfYiuhNI86e+p//Z"
3: Object
data: "data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wBDAAMCAgICAgMCAgIDAwMDBAYEBAQEBAgGBgUGCQgKCgkICQkKDA8MCgsOCwkJDRENDg8QEBEQCgwSExIQEw8QEBD/2wBDAQMDAwQDBAgEBAgQCwkLEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBD/wAARCAGQAyADASIAAhEBAxEB/8QAHwAAAQUBAQEBAQEAAAAAAAAAAAECAwQFBgcICQoL/8QAtRAAAgEDAwIEAwUFBAQAAAF9AQIDAAQRBRIhMUEGE1FhByJxFDKBkaEII0KxwRVS0fAkM2JyggkKFhcYGRolJicoKSo0NTY3ODk6Q0RFRkdISUpTVFVWV1hZWmNkZWZnaGlqc3R1dnd4eXqDhIWGh4iJipKTlJWWl5iZmqKjpKWmp6ipqrKztLW2t7i5usLDxMXGx8jJytLT1NXW19jZ2uHi4+Tl5ufo6erx8vP09fb3+Pn6VOV1ofMV009EQW+nRxYLnJ69Kt+WAR8v5VEl/aNy0w57+lWo57Nz8sqOfYiuhNI86e+p//Z"
```

(note these base64 strings are incomplete)

If you iterate with each over `images` then it will only iterate twice and ignore the base64 strings in the array. Any idea what causes this?

**Comment by ManuelDeLeon** [comment]
Make a repro.

**Comment by satyavh** [comment]
Will do, working on it. 

I think I found the core of the issue. When repeating over a template with a viewmodel attached, helpers created by that viewmodel are not scoped correctly (unlike v1), but only when setting viewmodel properties via onCreated or onRendered.

Each viewmodel overwrites the helpers of the previous template, so in my unfortunate case that results in helpers returning empty data for all templates.

**Comment by ManuelDeLeon** [comment]
I woudn't be surprised if it was the way around, v2 doing it right :)

I'll be happy to check it out once you have a repro.

**Comment by satyavh** [comment]
haha yes I've noticed v2 is more strict - which is def a good thing. 

here's the repro, I hope it's clear.
https://github.com/satyavh/viewmodel-each

btw, I noticed viewmodel explorer isn't compatible to v2? Or did I miss something?

**Comment by ManuelDeLeon** [comment]
Not near a computer but try renaming the template 'people' to 'peopleX'
(different than the array) and also pushing the values onCreated instead of
replacing the reactive array with the plain one (like you"re doing now)

**Comment by satyavh** [comment]
I've tried that (just committed), but doesn't change anything

**Comment by ManuelDeLeon** [comment]
The duplicates are there because you're adding each entry in the array again in the onCreated. Comment the onCreated and the duplicates are gone.

The button didn't work because of a bug when using Blaze helpers in nested `#each` blocks. This has been fixed in v2.5.3

btw, view model explorer doesn't work with v2. I'll work on a better one later on.

**Comment by satyavh** [comment]
Awesome!! Nice, we squashed a bug :-)

Looking forward to new viewmodel explorer, it's quite useful. Thanks a lot!
